src/argilla/client/feedback.py

```diff
@@ -672,6 +680,209 @@ def from_argilla(
             self.fetch_records()
         return self
 
+    def format_as(self, format: Literal["datasets"]) -> "Dataset":
+        """Formats the `FeedbackDataset` as a `datasets.Dataset` object.
+
+        Args:
+            format: the format to use to format the `FeedbackDataset`. Currently supported formats are:
+                `datasets`.
+
+        Returns:
+            The `FeedbackDataset.records` formatted as a `datasets.Dataset` object.
+
+        Raises:
+            ValueError: if the provided format is not supported.
+
+        Examples:
+            >>> import argilla as rg
+            >>> rg.init(...)
+            >>> dataset = rg.FeedbackDataset.from_argilla(name="my-dataset")
+            >>> huggingface_dataset = dataset.format_as("datasets")
+        """
+        if format == "datasets":
+            from datasets import Dataset, Features, Sequence, Value
+
+            dataset = {}
+            features = {}
+            for field in self.fields:
+                if field.settings["type"] not in FIELD_TYPE_TO_PYTHON_TYPE.keys():
+                    raise ValueError(
+                        f"Field {field.name} has an unsupported type: {field.settings['type']}, for the moment only the"
+                        f" following types are supported: {list(FIELD_TYPE_TO_PYTHON_TYPE.keys())}"
+                    )
+                features[field.name] = Value(dtype="string", id="field")
+                if field.name not in dataset:
+                    dataset[field.name] = []
+            for question in self.questions:
+                # TODO(alvarobartt): if we constraint ranges from 0 to N, then we can use `ClassLabel` for ratings
+                features[question.name] = Sequence(
+                    {
+                        "user_id": Value(dtype="string"),
+                        "value": Value(dtype="string" if question.settings["type"] == "text" else "int32"),
+                    },
+                    id="question",
+                )
+                if question.name not in dataset:
+                    dataset[question.name] = []
+            features["external_id"] = Value(dtype="string", id="external_id")
+            dataset["external_id"] = []
+
+            for record in self.records:
+                for field in self.fields:
+                    dataset[field.name].append(record["fields"][field.name])
+                for question in self.questions:
+                    dataset[question.name].append(
+                        [
+                            {"user_id": r["user_id"], "value": r["values"][question.name]["value"]}
+                            for r in record["responses"]
+                        ]
+                        or None
+                    )
```

Comment:
Maybe makes sense to move this code block into a private method `_format_as_datasets`

Reply:
As per the comment below: 

> I left it like format_as to basically allow users to e.g. pull from Argilla, format it as they want (i.e. datasets), and then use it, without having to push anything, just to format the current dataset records in the format they want. The idea is to extend the format_as method to support pandas, numpy, etc.